### ISSUE TITLE
default_sort field that' is also in 'translate' list

### ISSUE THREAD

**Issue by sparkcom** [tractorcow-farm/silverstripe-fluent#606]
If a Dataobject has a field that's the default sort, for example 

`class Object extends DataObject {
private static $db  = ['Title' =>'Varchar'];

private static $default_sort ='Title ASC'
}`

And then when a Grieldfield tried to bring it up,  the sql will try to do 'Sort by DataObject.Title ASC'. But the problem is, the field 'DataObject.Title' is NOT in the select, so there will be sql error.   If  default_sort is 'Created', which is not in 'translate' list, there is no error.

**Comment by tractorcow** [comment]
Looks like a serious issue; We need to make sure the sorted columns are localised too.

**Comment by tractorcow** [comment]
Hm the sorting columns are ment to be localised. There is code in there to handle it.

You need to use quoted columns though. E.g. `private static $default_sort ='"Title" ASC';` for it to rewrite.

**Comment by tractorcow** [comment]
@sparkcom can you please adjust and test and confirm if this fixes your issue?

**Comment by cwchong** [comment]
> Hm the sorting columns are ment to be localised. There is code in there to handle it.
> 
> You need to use quoted columns though. E.g. `private static $default_sort ='"Title" ASC';` for it to rewrite.

This works for me on SS4.7.0 and fluent 4.x-dev 👍

**Comment by AlphaCactus** [comment]
> 
> 
> > Hm the sorting columns are ment to be localised. There is code in there to handle it.
> > You need to use quoted columns though. E.g. `private static $default_sort ='"Title" ASC';` for it to rewrite.
> 
> This works for me on SS4.7.0 and fluent 4.x-dev 👍

Double quote wrapping the sort column name works for me as well.  SS 4.7.0 and fluent 4.5.1

**Comment by beerbohmdo** [comment]
This issue also happens with the array syntax:

```php
private static array $default_sort = [
   'Title' => 'asc',
];
```

This results into `ORDER BY "DataObject"."Title" ASC`

Quoting the key seems to work here too :raised_eyebrow: .

**Comment by tractorcow** [comment]
Yep, that's one of the long-standing quirks of silverstripe. Some things only really work if you strictly quote everything, but it's not enforced anywhere.

**Comment by beerbohmdo** [comment]
The strange part is that at the Point when the FluentExtension runs it is stil `['Title' => 'asc']`. But in the resulting query it is correctly quoted.
